src/detect-pcre.c

```diff
@@ -70,15 +71,28 @@
 static int pcre_match_limit = 0;
 static int pcre_match_limit_recursion = 0;
 
-static pcre *parse_regex;
-static pcre_extra *parse_regex_study;
-static pcre *parse_capture_regex;
-static pcre_extra *parse_capture_regex_study;
+static DetectParseRegex parse_regex;
+static DetectParseRegex parse_capture_regex;
 
 #ifdef PCRE_HAVE_JIT
 static int pcre_use_jit = 1;
 #endif
 
+/* \brief Helper function for using pcre_exec with/without JIT
+ */
+static int DetectPcreExec(DetectParseRegex *regex, const char *str, const size_t strlen,
+                            int start_offset, int options,  int *ovector, int ovector_size)
+{
+#ifdef PCRE_HAVE_JIT_EXEC
+    if (pcre_use_jit)
```

Comment:
I suppose this could also be checked by seeing if ```regex->jit_stack``` is non-NULL?

Reply:
Yes ... if fallback is supported (prev comment).